flink-core/src/main/java/org/apache/flink/types/variant/BinaryVariantInternalBuilder.java

```diff
@@ -290,7 +290,7 @@ public void appendTimestamp(long microsSinceEpoch) {
     public void appendFloat(float f) {
         checkCapacity(1 + 4);
         writeBuffer[writePos++] = primitiveHeader(FLOAT);
-        writeLong(writeBuffer, writePos, Float.floatToIntBits(f), 8);
+        writeLong(writeBuffer, writePos, Float.floatToIntBits(f), 4);
```

Comment:
Hi,
Am I missing something? Long is 8 bytes and it is a writeLong.  Could it be that  `checkCapacity(1 + 4);` is incorrect and it should be  `checkCapacity(1 + 8);` Line 286 is a write long with 8 bytes